src/parser/mod.rs

```diff
@@ -2328,19 +2327,25 @@ impl<'a> Parser<'a> {
         }
     }
 
-    /// Bigquery specific: Parse a struct literal
     /// Syntax
     /// ```sql
-    /// -- typed
+    /// -- typed, specific to bigquery
     /// STRUCT<[field_name] field_type, ...>( expr1 [, ... ])
     /// -- typeless
     /// STRUCT( expr1 [AS field_name] [, ... ])
     /// ```
-    fn parse_bigquery_struct_literal(&mut self) -> Result<Expr, ParserError> {
-        let (fields, trailing_bracket) =
-            self.parse_struct_type_def(Self::parse_struct_field_def)?;
-        if trailing_bracket.0 {
-            return parser_err!("unmatched > in STRUCT literal", self.peek_token().location);
+    fn parse_struct_literal(&mut self) -> Result<Expr, ParserError> {
+        let mut fields = vec![];
+        // Typed struct syntax is only supported by BigQuery
+        // https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#typed_struct_syntax
+        if self.dialect.supports_typed_struct_syntax() {
```

Comment:
Oh actually I'm not sure i follow the changes entirely, it seems Bigquery struct syntax is a superset of databricks struct? If so I'm thinking we ideally wouldnt need the extra `self.dialect.supports_typed_struct_syntax()` method, only guarding on `supports_struct_literal` should be enough?

It would technically mean that the typed syntax gets accepted by the databricks dialect but thats fine for the parser behavior

Reply:
You are right, BigQuery struct is a superset of the Databricks struct. I wasn't sure if `supports_typed_struct_syntax` is too much or not. It sounds odd though that we would accept the syntax in Databricks even if it is invalid for this dialect. But I'm happy to change it if that is acceptable behavior for the parser.

Reply:
done as suggested.